### PR TITLE
fix: skip dropout in eval for flash_attn in various models

### PR DESCRIPTION
**Pull request by fdschmidt93**
# What does this PR do?

For `M2M100FlashAttention2` was not skipped when `model.eval()` was called, which of course quite significantly affects model outputs. This PR fixes the issue by setting dropout to `0.0` when `M2M100FlashAttention2.training == False`.

E: It turns out this affected all models that copied their FA2 implementation from Bart.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by vasqu** [comment]
I have no idea why the FA2 implementation doesn't use a `# Copied from` statement from bart here tbh. Nevertheless, it also reveals this dropout issue in any bart related model / attn implementation. For reference: https://github.com/huggingface/transformers/blob/570c89625b329a2d33ecbfb972a9aae015535471/src/transformers/models/bart/modeling_bart.py#L403

Might be worth to create a superceding PR for bart and add the missing "copied from" for m2m_100 over here.

cc @amyeroberts @ArthurZucker

**Comment by vasqu** [comment]
Thx for the quick fix! :)

You can run `make fix-copies` and `make style` which handle the "copied from" mechanism and linting respectively. 

And just like hf bot says, you can push an empty commit at the end with `[run-slow] bart, m2m_100` (I doubt we need to test the other copied from models but correct me if I'm wrong 👀). A maintainer will handle those slow runs when they have time.

**Comment by fdschmidt93** [comment]
Thanks! Hopefully everything addressed now :)

**Review by ArthurZucker — APPROVED** [review]
thanks! that's quite a bad bug indeed

**Comment by ArthurZucker** [comment]
Thanks @vasqu as well for landing a hand!

**Comment by ArthurZucker** [comment]
🤗 `
